tests/test_resource_warning.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+import time
+import typing
+import warnings
+from datetime import datetime
+
+import pytest
+
+import aiohttp
+
+last_warning: typing.Optional[str] = None
+
+
+def print_warnings(message, category, filename, lineno, file=None, line=None):
+    global last_warning
+    text = f'{datetime.now()} | ' \
+           f'{filename}:{lineno} {category.__name__:s}:{message}'
+    print(text)
+    # These are warnings in the 4.0a and these prevent execution of the test
+    if "RuntimeWarning:coroutine 'noop' was never awaited" in text:
+        return None
+    last_warning = text
+
+
+@pytest.fixture()
+def warning_setup(monkeypatch):
+    # setup warning
+    monkeypatch.setattr(warnings, 'showwarning', print_warnings)
+    warnings.simplefilter('default')
+
+
+async def _resource_warning():
+    timeout = 3 * 60
+
+    client = aiohttp.ClientSession()
+
+    request_interval = 30
+    request_increase = 15
+    next_request = 0
+
+    end = time.time() + timeout
+    while last_warning is None:
+        now = time.time()
+        if now > end:
+            break
+
+        if now > next_request:
+            async with client.get('https://heise.de') as resp:
```

Comment:
It's better to use a local server. Here's fixtures that will help you to set up a trusted TLS context while still using real-world SSL: https://github.com/aio-libs/aiohttp/commit/c180800a4c90dc123d05311edbec92a3a82d6317#diff-5d43d8b7848138232c802e32ab023dd3R271-R272

Reply:
How would I use the fixture?

Reply:
Add it as an arg into fixtures that set up a web server and a client and pass as `ssl`:
>    server = await aiohttp_server(app, ssl=**ssl_ctx**)
>    conn = aiohttp.TCPConnector(ssl=**client_ssl_ctx**)

Reply:
I'm sorry I don't know how to setup a server and use it.
Can't we leave it like this, it seems to be working fine.

Reply:
Tests should work offline. Imagine you're traveling w/o any internet connectivity. You should be able to run this on your laptop.

Reply:
If you could make a suggestion I'd gladly copy paste it.

Reply:
Actually, you don't have to manually copy suggestions, it's better to click `Commit suggestion` from the GitHub UI because this way it automatically preserves authorship.